integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py

```diff
@@ -86,6 +91,27 @@ def __init__(
         self.backend: Optional[EmbedderBackend] = None
         self._initialized = False
 
+    def _url_validation(self, api_url):
+        ## Making sure /v1 in added to the url, followed by infer_path
+        result = urlparse(api_url)
+        expected_format = "Expected format is 'http://host:port'."
+
+        if api_url == _DEFAULT_API_URL:
+            return api_url
+        if result.path:
+            normalized_path = result.path.strip("/")
+            if normalized_path == "v1":
+                pass
+            elif normalized_path in ["v1/embeddings"]:
+                warn_msg = f"{expected_format} Rest is ingnored."
+                warnings.warn(warn_msg, stacklevel=2)
+            else:
+                err_msg = f"Base URL path is not recognized. {expected_format}"
+                raise ValueError(err_msg)
+
+        base_url = urlunparse((result.scheme, result.netloc, "v1", "", "", ""))
+        return base_url
```

Comment:
Looks like this code is mostly the same for all components? Let's move it into a `haystack_integrations.util.nvidia` module.

Reply:
Yes will do that

Reply:
@shadeMe can you take a look now?